### DISCUSSION TITLE
AsyncLocalStorage with Pages Router

### DISCUSSION THREAD

**Discussion by SalmenBejaoui** [vercel/next.js#66841] [Help]
### Summary

I was wondering if it is possible to use AsyncLocalStorage API per Pages Router to pass data from getServerSideProps into lower components without impacting __NEXT_DATA__ size.

**Comment by icyJoseph** [comment]
Wouldn't this cause issues during hydration?

**Reply by SalmenBejaoui** [reply]
I think so. This is why I asked the question. As I'm not able to migrate to App Router, I was thinking if there is any solution to data between component without adding it into NEXT_DATA to reduce HTML size.
